__tests__/app.test.js

```diff
@@ -0,0 +1,50 @@
+const db = require(`${__dirname}/../db/connection.js`);
+const { app } = require(`${__dirname}/../src/app.js`);
+const data = require(`${__dirname}/../db/data/test-data/index.js`);
+const seed = require(`${__dirname}/../db/seeds/seed.js`);
+const request = require("supertest");
+
+beforeEach(() => {
+  return seed(data);
+});
+
+afterAll(() => db.end());
+
+describe("GET /api/categories", () => {
+  it("200: Should respond with an array of category objects", () => {
+    return request(app)
+      .get("/api/categories")
+      .expect(200)
+      .then(({ body }) => {
+        const { categories } = body;
+        expect(categories).toBeInstanceOf(Array);
+        expect(categories).toHaveLength(4);
+        categories.forEach((category) => {
+          expect(category).toBeInstanceOf(Object);
+        });
+      });
+  });
+  it("200: response objects should have expected shape", () => {
+    return request(app)
+      .get("/api/categories")
+      .expect(200)
+      .then(({ body }) => {
+        const { categories } = body;
+        categories.forEach((category) => {
+          expect(category).toHaveProperty("slug", expect.any(String));
```

Comment:
I would suggest keeping these assertions in the above it block, if for any reason we aren't getting anything back from this second request or the test is written incorrectly the forEach could not run properly and therefore not iterate over anything and this could create a false pass. It's therefore best to keep this with the previous expect statements so we know that the response both has a particular length and we know what keys it includes.

Reply:
On it boss! 😄